### ISSUE TITLE
Updating from 5.2.0 to 5.2.1 broke hover behaviour in nav-items and dropdown items

### ISSUE THREAD

**Issue by phev8** [twbs/bootstrap#37105]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

This being a patch release, we did not expect breaking changes, but apparently some behaviour broke hover styles for nav-link / btn and dropdown items (used from react-bootstrap). 

I checked the changelog, but I could not identify what the related change is and where we should start looking. Is the changelog maybe incomplete?

Our app's behaviour before update:

https://user-images.githubusercontent.com/13927969/189116540-1531cd88-c3ef-42a4-a45f-f971fa3bbdc7.mov

After update: 

https://user-images.githubusercontent.com/13927969/189116563-c887a071-8236-4caf-af78-c76e6a3c8b4a.mov

(The only change between the two was to update bootstrap from 5.2.0 to 5.2.1)

The same update causes issues when using the Dropdown items from react bootstrap:

Before:
https://user-images.githubusercontent.com/13927969/189118800-9b7e6864-0251-4ed7-9314-9758503bfedd.mov

After (The slight background color change not there anymore):
https://user-images.githubusercontent.com/13927969/189118806-fc2618e8-24a1-4b79-ab2b-bc0f6202e6ea.mov

### Reduced test cases

Our navbar related style overrides are here:
https://github.com/coneno/case-web-ui/blob/master/src/scss/helpers/navbar.scss

And the navbar item is implemented here:
https://github.com/coneno/case-web-app-core/blob/master/src/components/navbar/NavbarComponents/NavbarItem.tsx

(with class names "btn nav-link nav-link-height" attached)

What change from the release might cause the displayed issue? We can make an isolated example to reproduce then, but at the moment, it's quite hard to pinpoint where this behaviour change might come from for us.

Thank you!

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome, Safari

### What version of Bootstrap are you using?

5.2.1

**Comment by julien-deramond** [comment]
When [comparing v5.2.0 and v5.2.1](https://github.com/twbs/bootstrap/compare/v5.2.0...v5.2.1) I don't see at first glance in our (S)CSS modifications what could have generated this issue. Maybe things that have been modified in [buttons.scss](https://github.com/twbs/bootstrap/compare/v5.2.0...v5.2.1#diff-02f9f9f177b330ca219b9b681289f921a06b7ffa9251d48465bf9a16def6c270) :shrug: 
Please try to create a minimal reproductible example in CodePen or StackBlitz. If that's not possible I'll try to find time to compile your GitHub repository.

**Comment by phev8** [comment]
Thank you for the quick response. I am still a little bit in the dark, but noticed the following:

Having a simple `<button class="btn btn-primary">Test</button>`, has different behaviour, where in previous version, we had a nice outline around the button in Chrome when clicked, with new version, we have nothing.

Here is an example, where if you change the bootstrap import to the latest, in Chrome, it does not show the outline anymore.

https://codepen.io/phev8/pen/wvjMyXL

Expected (5.2.0)
<img width="139" alt="Screenshot 2022-09-08 at 15 46 19" src="https://user-images.githubusercontent.com/13927969/189138947-2aea37a1-a35d-4757-8fae-47904de72cff.png">

**Comment by phev8** [comment]
I think my issues might be related that these variable overrides are not working as before:
```
$navbar-text-font-size: 1.125rem !default;
$navbar-text-padding-top: 10px !default;
$nav-link-color: white !default;
$nav-link-font-size: 1.125rem !default;
$nav-link-hover-color: white !default;
$nav-tabs-border-width: 0px !default;
$navbar-toggler-border-radius: 0px !default;
$nav-tabs-link-active-bg: map-get($theme-colors, "secondary") !default;
```

I could not figure it out yet, how can I reproduce this variable overrides with CodePen, if someone has quick example, of how to use the scss variables there, I am glad for a quick hint :-)

**Comment by phev8** [comment]
So I have a minimalistic case, which seems to be related:
```
<div class="bg-primary">
  <a
    class="nav-link"
  >Link 1</a>
  <a
    class="text-body nav-link active"
  >Link 2</a>
  <a
    class="nav-link"
  >Link 3</a>

</div>
```

This with the previously mentioned variable overrides (custom theme), when using 5.2.0 it compiles to:
<img width="589" alt="Screenshot 2022-09-08 at 16 15 45" src="https://user-images.githubusercontent.com/13927969/189146027-3811d2b3-a2bb-41bc-bd31-82e4f30738c9.png">

With 5.2.1 it compiles to:
<img width="594" alt="Screenshot 2022-09-08 at 16 14 35" src="https://user-images.githubusercontent.com/13927969/189146139-6c8e227e-d597-415e-af14-e26d5a413662.png">

**Comment by julien-deramond** [comment]
Created two StackBlitz projects to check your last minimalistic case including your variable overrides:
- [StackBlitz #37105 (v5.2.0)](https://stackblitz.com/edit/github-4yh9ha-chswdh?file=index.html)
- [StackBlitz #37105 (v5.2.1)](https://stackblitz.com/edit/github-4yh9ha?file=index.html)

Maybe I've made some errors creating it but the rendering seems to be the same (but different than in your screenshots).

**Comment by phev8** [comment]
Thank you for the examples. 

It's very strange, since with my local installation I had 5.2.0 installed before and it was working... but not on the StackBlitz example, I had to go back to 5.1.3 then it's how we had it. 
- [StackBlitz #37105 (v5.1.3)](https://stackblitz.com/edit/github-4yh9ha-7sqeqg?file=scss/styles.scss)

Includes the example how you created, using version 5.1.3, and "bootstrap/maps" commented out (not there yet). 
Produces the result:

<img width="695" alt="Screenshot 2022-09-08 at 21 49 26" src="https://user-images.githubusercontent.com/13927969/189212926-0733c67d-6baf-49b6-af8a-5f64aa4c4d9b.png">

In the examples, from version 5.2.0 the variable overrides for the nav-link do not seem to work the way we had there anymore. Did we miss some deprecations there?

**Comment by julien-deramond** [comment]
Created [an updated StackBlitz based on my previous example on v5.2.1](https://stackblitz.com/edit/github-4yh9ha-jjame1?file=index.html,scss%2Fstyles.scss):
* in `styles.css` I moved the overrides (sorry for that they were not at the good place)
* in the HTML I changed the structure to have a `.nav` as presented in our [Vertical example](https://getbootstrap.com/docs/5.2/components/navs-tabs/#vertical). It is needed to define the white link color. Reference: https://github.com/twbs/bootstrap/blob/23e50829f958ea1d741d63e2781716be037e4644/scss/_nav.scss#L12

Is that useful for you?

**Comment by phev8** [comment]
Thank you very much, it is useful!

Through this, I also noticed what my actual problem is and how to fix it after upgrade. The only question remaining, if this is intentional:
- I sometimes like to use <button> instead of <a> tags, when the action is not really a link /or it's an internal router link/, otherwise the "accessibility" starts to complain that I'm using incorrect <a> tags... (E.g., in case, from the nav menu, a dialog would open, I have no href). I am no expert on the topic, but so far, using simply a button helped.
- However built in browser button style... you know. So I used the bootstrap "btn" class to clean that. 

With previous versions, attaching the "btn" class worked fine for me, with 5.2.0, when nav-link is active and has btn class, now I also have to extra remove the border. With latest update however, when nav-link also has the btn tag, the btn styling seems to have priority or modifies the prefix, I don't know. So the nav-link color is not applied in hover state any more. 

Updated your example with the case here:
- [StackBlitz example with nav-link buttons](https://stackblitz.com/edit/github-4yh9ha-vbuubn?file=index.html)
